web-service/helpers.py

```diff
@@ -132,42 +130,48 @@ def SortByDistanceCmp(a, b):
 
 ################################################################################
 
-def FetchAndStoreUrl(siteInfo, url):
+def GetSiteInfoForUrl(url, rssi=None, txpower=None, force_update=False):
+    siteInfo = models.SiteInformation.get_by_id(url)
+
+    if force_update or siteInfo is None:
+        siteInfo = FetchAndStoreUrl(siteInfo, url, rssi, txpower, force_update)
+
+    return siteInfo
+
+################################################################################
+
+def FetchAndStoreUrl(siteInfo, url, rssi=None, txpower=None, force_update=False):
     # Index the page
     try:
-        result = urlfetch.fetch(url, validate_certificate = True)
+        headers = {}
+        if rssi is not None and txpower is not None:
+            headers['X-PhysicalWeb-Rssi'] = rssi
+            headers['X-PhysicalWeb-TxPower'] = txpower
+
+        result = urlfetch.fetch(url,
+                                follow_redirects=False,
```

Comment:
Note, we no longer follow redirect ever.

Previously we would follow redirects to the end, and then make another request to the initial redirector only for a subset of urls.  This way, we don't do any custom handling, nor make needless fetches.